moveit_py/src/moveit/moveit_core/collision_detection/collision_common.cpp

```diff
@@ -0,0 +1,133 @@
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Peter David Fagan */
+
+#include "collision_common.h"
+
+namespace moveit_py
+{
+namespace bind_collision_detection
+{
+void init_collision_request(py::module& m)
+{
+  py::module collision_detection = m.def_submodule("collision_detection");
+
+  py::class_<collision_detection::CollisionRequest>(collision_detection, "CollisionRequest", R"(
+      Representation of a collision checking request.
+      )")
+
+      .def(py::init<>())
+
+      .def_readwrite("joint_model_group_name", &collision_detection::CollisionRequest::group_name,
+                     R"(
+                     str: The group name to check collisions for (optional; if empty, assume the complete robot)
+                     )")
+
+      .def_readwrite("distance", &collision_detection::CollisionRequest::distance,
+                     R"(
+                     bool: If true, compute proximity distance.
+                     )")
+
+      .def_readwrite("cost", &collision_detection::CollisionRequest::cost,
+                     R"(
+                     bool: If true, a collision cost is computed.
+                     )")
+
+      .def_readwrite("contacts", &collision_detection::CollisionRequest::contacts,
+                     R"(
+                     bool: If true, compute contacts.
+                     )")
+
+      .def_readwrite("max_contacts", &collision_detection::CollisionRequest::max_contacts,
+                     R"(
+                     int: Overall maximum number of contacts to compute.
+                     )")
+
+      .def_readwrite("max_contacts_per_pair", &collision_detection::CollisionRequest::max_contacts_per_pair,
+                     R"(
+                     int: Maximum number of contacts to compute per pair of bodies (multiple bodies may be in contact at different configurations).
+                     )")
+
+      .def_readwrite("max_cost_sources", &collision_detection::CollisionRequest::max_cost_sources,
+                     R"(
+                     int: When costs are computed, this value defines how many of the top cost sources should be returned.
+                     )")
+
+      // TODO (peterdavidfagan): define is_done as function call.
+      //.def_readwrite("is_done", &collision_detection::CollisionRequest::is_done,
+      //               R"(
+      //               )")
```

Comment:
For this PR as well?

Reply:
I would prefer to keep this for the future.